ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,371 @@
+    @objc public init(with contentViewController: UIViewController) {
+        self.contentViewController = contentViewController
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// A scroll view in `contentViewController`'s view hierarchy.
+    /// Provide this to ensure the bottom sheet pan gesture recognizer coordinates with the scroll view to enable scrolling based on current bottom sheet position and content offset.
+    @objc open var hostedScrollView: UIScrollView?
+
+    /// Indicates if the bottom sheet is expandable.
+    @objc open var isExpandable: Bool = true {
+        didSet {
+            if isExpandable != oldValue {
+                resizingHandleView.isHidden = !isExpandable
+                panGestureRecognizer.isEnabled = isExpandable
+                move(to: .collapsed, animated: false)
+            }
+        }
+    }
+
+    /// Indicates if `preferredContentSize` of `contentViewController` should be respected.
+    /// Regardless of the value, the expanded height is limited by the height of `BottomSheetController`'s root view.
+    @objc open var respectsPreferredContentSize: Bool = false {
+        didSet {
+            if respectsPreferredContentSize != oldValue {
+                updateBottomSheetHeightConstraints()
+            }
+        }
+    }
+
+    /// Fraction of the available area that the bottom sheet shold take up in the expanded position.
+    ///
+    /// Ignored when `respectsPreferredContentSize` is set to `true`
+    @objc open var expandedHeightFraction: CGFloat = 1.0 {
+        didSet {
+            if expandedHeightFraction != oldValue && !respectsPreferredContentSize {
+                updateBottomSheetHeightConstraints()
+            }
+        }
+    }
+
+    /// Height of the top portion of the content view that should be visible when the bottom sheet is collapsed.
+    @objc open var collapsedContentHeight: CGFloat = Constants.defaultCollapsedContentHeight
+
+    /// The object that acts as the delegate of the bottom sheet.
+    @objc open weak var delegate: BottomSheetControllerDelegate?
+
+    // MARK: - View loading
+
+    public override func loadView() {
+        view = BottomSheetPassthroughView()
+
+        addChild(contentViewController)
+        contentViewController.didMove(toParent: self)
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(bottomSheetView)
+
+        NSLayoutConstraint.activate([
+            bottomSheetView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            bottomSheetView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bottomSheetOffsetConstraint
+        ])
+        updateBottomSheetHeightConstraints()
```

Comment:
currently I am seeing a bug when expanded and rotating, the height of the bottomsheet on landscape isn't calculated correctly

Reply:
Solved by doing the safe thing of collapsing the sheet on any root view size changes. That way the sheet can't find itself in an invalid offset when the height changes in unexpected ways.

Reply:
although, isn't it weird that when you have it expanded and you rotate, it will collapse? shouldn't it stay expanded?

Reply:
Good point, and turns it this also collapses it when only the width changes… I’ll try to preserve the expanded/collapsed state.

Reply:
Made a change to move to the correct offset for the new height.